File: scripts/audit-production-recipes.mjs
import { getDb } from "../src/lib/db.ts";
import { parseValues } from "../src/lib/recipe-map.ts";
import {
  formatTime,
  totalMinutes,
  countedHeatMinutes,
} from "../src/lib/recipe-utils.ts";
import { publicRestMinutes, riseHoursFromExtras } from "../src/lib/recipe-timing.ts";

const prisma = getDb();

function imageHost(url) {
  if (!url) return "empty";
  if (url.startsWith("/")) return "local-path";
  try {
    return new URL(url).hostname;
  } catch {
    return "invalid-url";
  }
}

function extrasFromRow(row, values) {
  const typeFields = row.type?.fields || [];
  return typeFields.map((f) => ({ key: f.key, value: values[f.key] }));
}

const rows = await prisma.recipe.findMany({
  where: { status: "published" },
  include: {
    categories: { include: { category: true } },
    type: { include: { fields: true } },
  },
  orderBy: { title: "asc" },
});

for (const row of rows) {
  const values = parseValues(row.values);
  const extras = extrasFromRow(row, values);
  const recipeLike = {
    prepMinutes: values.prepMinutes ?? 0,
    cookMinutes: values.cookMinutes ?? 0,
    bakeMinutes: values.bakeMinutes ?? 0,
    restMinutes: values.restMinutes ?? 0,
    extras: extras.map((e) => ({
      key: e.key,
      label: e.key,
      kind: "number",
      value: e.value,
    })),
  };
  const riseHours = riseHoursFromExtras(recipeLike);
  const gallery = Array.isArray(values.gallery) ? values.gallery : [];

  console.log(
    JSON.stringify({
      title: row.title,
      slug: row.slug,
      image: values.image || "",
      imageAlt: values.imageAlt || "",
      imageHost: imageHost(String(values.image || "")),
      gallery: gallery.slice(0, 8),
      prepMinutes: values.prepMinutes,
      cookMinutes: values.cookMinutes,
      bakeMinutes: values.bakeMinutes,
      restMinutes: values.restMinutes,
      riseHours,
      riseHoursField: values.riseHours,
      cardTotal: formatTime(totalMinutes(recipeLike)),
      heat: countedHeatMinutes(recipeLike),
      publicRest: publicRestMinutes(recipeLike),
      servings: values.servings,
      servingsUnit: values.servingsUnit,
      excerpt: row.excerpt,
    }),
  );
}

console.log(JSON.stringify({ publishedCount: rows.length }));
await prisma.$disconnect();

File: scripts/audit-production-recipes.py
import sqlite3
import json
from pathlib import Path

DB = Path(__file__).resolve().parents[1] / "prisma" / "dev.db"
conn = sqlite3.connect(DB)
rows = conn.execute(
    """
    SELECT slug, title, excerpt, "values", status
    FROM Recipe
    ORDER BY title
    """
).fetchall()

published = [r for r in rows if r[4] == "published"]
print(f"total={len(rows)} published={len(published)}")

for slug, title, excerpt, values_raw, status in published:
    try:
        values = json.loads(values_raw or "{}")
    except json.JSONDecodeError:
        values = {}
    image = values.get("image", "")
    print(
        json.dumps(
            {
                "slug": slug,
                "title": title,
                "image": image,
                "imageAlt": values.get("imageAlt", ""),
                "gallery": values.get("gallery", [])[:5],
                "prepMinutes": values.get("prepMinutes"),
                "cookMinutes": values.get("cookMinutes"),
                "bakeMinutes": values.get("bakeMinutes"),
                "restMinutes": values.get("restMinutes"),
                "riseHours": values.get("riseHours"),
                "servings": values.get("servings"),
                "servingsUnit": values.get("servingsUnit"),
                "excerpt": excerpt,
            },
            ensure_ascii=False,
        )
    )

conn.close()
